### ISSUE TITLE
From version 1.5.0a further excavator supports only NiceHash stratum servers. 

### ISSUE THREAD

**Issue by MichaelMiner** [nicehash/excavator#228]
Excavator dead

**Comment by streaml1ne555** [comment]
Yea, lame choice there fellas. Put a dev fee on if you want to make some money from non-Nice Hash users.

**Comment by apiontek** [comment]
It's also not generally faster than 1.4.4a for neoscrypt, lyra2v2, or nist5, and not faster than dstm, ethminer, or xmrig/xmr-stak for equihash, ethash, and cryptonight ... so until there's some compelling reason to use excavator 1.5+, just keep using 1.4.4a, nobody's stopping you.

Implementing a dev fee would both cost time and energy, and probably eliminate the edge in speed ... really it's in Nicehash's best interest to not worry about people mining at other pools. If excavator truly excels over other miners in valuable algorithms, you'll want to mine with them or else miss out on the profits. 

And I say all that as someone who doesn't generally mine with them.

I mean, we may as well complain that it's not open source, which to me would seem a much more valid and weighty criticism.

**Comment by AlwaysCandr** [comment]
Really uncool. I would agree with putting 1-2% fee for non-nicehash pools. But removing the ability for other pools... That's just nasty.

**Comment by apiontek** [comment]
Well frankly, this finally forced me to do some real testing and find that excavator isn't really the best at either lyra2v2 or neoscrypt.

I have 4 gtx 1070s. I ran excavator 1.4.4a on them all at ahashpool for an hour, sampling the hashrate reported by the pool every five minutes. I did this for both neoscrypt and lyra2v2. I then did the same with KlausT's ccminer.

For lyra2v2, the pool-reported hashrate was consistently less than half of what excavator was reporting, and half what KlausT's ccminer got.

For neoscrypt, the pool-reported hashrate was consistently 20% less than what excavator was reporting, and 15% less than what KlausT's ccminer got.

So, while excavator looks good and it seems frustrating to not be able to use future versions outside of nicehash ... I'm no longer convinced it was ever worth using excavator over other miners in the first place.

Maybe there's room for me doing the same tests over again on different pools in case there are issues with stratum protocols at those pools, and doing them at nicehash as well, but this was enough for me.

**Comment by apiontek** [comment]
I also tested ewbf (with 0 fee option), dstm, and excavator at flypool and found that while I've been talking up excavator (on linux) for a while because its reported hashrate is higher than dstm's when you subtract dstm's fee (on linux) ...

Instead I found that on the pool side, not only is excavator weaker than ewbf, miraculously on my machines, dstm achieved lower rates (and fewer shares) than ewbf (with 0 fee).

I also tested ethash at ethermine.org and found that for all but one of my machines, the latest ethminer is consistently better than claymore or excavator. On the one outlier machine, claymore was better.

For myself, I see no reason to care further about excavator, and even when I want to mine at nicehash, while I might test again just to double-check, it seems unwise to use their own miner...
